File: 1566. Detect Pattern of Length M Repeated K or More Times/Solution.java
class Solution {
    public boolean containsPattern(int[] arr, int m, int k) {
        int goal = (k - 1) * m, count = 0;
        for (int i = 0; i + m < arr.length; i++) {
            if (arr[i] == arr[i + m]) {
                if (++count == goal) {
                    return true;
                }
            } else {
                count = 0;
            }
        }
        return false;
    }
}

File: 1566. Detect Pattern of Length M Repeated K or More Times/Solution.py
from typing import List


class Solution:
    def containsPattern(self, arr: List[int], m: int, k: int) -> bool:
        goal, cnt = (k - 1) * m, 0
        for i in range(len(arr) - m):
            cnt = cnt + 1 if arr[i] == arr[i + m] else 0
            if cnt == goal:
                return True
        return False
